source/Irrlicht/CBAWMeshWriter.cpp

```diff
@@ -340,18 +331,33 @@ namespace irr {namespace scene {
 
 		if (_encrypt)
 		{
-			fcrypt_ctx cryptCtx;
-			unsigned char dummy[10];
-			const unsigned char* pwd = (_ctx.pwdVer[0] == 0xff && _ctx.pwdVer[1] == 0xff) ? /* "no password" */ (const unsigned char*)"hejkahejkahejkaa" : _ctx.props->encryptionPassPhrase;
-			fcrypt_init(1, pwd, 16, _ctx.props->initializationVector, dummy, &cryptCtx);
-			fcrypt_encrypt(reinterpret_cast<unsigned char*>(data), compressedSize, &cryptCtx);
-			comprType |= core::Blob::EBCT_AES128_GCM;
-			fcrypt_end(dummy, &cryptCtx);
+			const size_t encrSize = core::BlobHeaderV0::calcEncSize(compressedSize);
+			void* in = malloc(encrSize);
+			memset(((uint8_t*)in) + (compressedSize-16), 0, 16);
+			memcpy(in, data, compressedSize);
```

Comment:
if you used EVP padding properly inside core::runAes128gcm, you wouldn't need a redundant memcpy
(hell even if you can't make that work, round up the `data` malloc to 16 bytes and you'll never need padding)